alicloud/service_alicloud_ddosbgp.go

```diff
@@ -0,0 +1,112 @@
+package alicloud
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ddosbgp"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+type DdosbgpService struct {
+	client *connectivity.AliyunClient
+}
+
+func (s *DdosbgpService) DescribeDdosbgpInstance(id string) (v ddosbgp.Instance, err error) {
+	request := ddosbgp.CreateDescribeInstanceListRequest()
+	request.RegionId = s.client.RegionId
+	request.DdosRegionId = s.client.RegionId
+	request.InstanceIdList = "[\"" + id + "\"]"
+	request.PageNo = "1"
+	request.PageSize = "10"
+
+	raw, err := s.client.WithDdosbgpClient(func(ddosbgpClient *ddosbgp.Client) (interface{}, error) {
+		return ddosbgpClient.DescribeInstanceList(request)
+	})
+
+	if err != nil {
+		if IsExceptedErrors(err, []string{DdosbgpInstanceNotFound}) {
+			return v, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+
+		return v, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	response, _ := raw.(*ddosbgp.DescribeInstanceListResponse)
+	if len(response.InstanceList) == 0 || response.InstanceList[0].InstanceId != id {
+		return v, WrapErrorf(Error(GetNotFoundMessage("Ddosbgp Instance", id)), NotFoundMsg, ProviderERROR)
+	}
+
+	v = response.InstanceList[0]
+	return
+}
+
+func (s *DdosbgpService) DescribeDdosbgpInstanceSpec(id string, region string) (v ddosbgp.InstanceSpec, err error) {
+	request := ddosbgp.CreateDescribeInstanceSpecsRequest()
+	request.InstanceIdList = "[\"" + id + "\"]"
+	request.DdosRegionId = region
+	request.RegionId = region
+
+	raw, err := s.client.WithDdosbgpClient(func(ddosbgpClient *ddosbgp.Client) (interface{}, error) {
+		return ddosbgpClient.DescribeInstanceSpecs(request)
+	})
+
+	if err != nil {
+		if IsExceptedErrors(err, []string{DdosbgpInstanceNotFound, InvalidDdosbgpInstance}) {
+			return v, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+
+		return v, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	resp, _ := raw.(*ddosbgp.DescribeInstanceSpecsResponse)
+	if len(resp.InstanceSpecs) == 0 || resp.InstanceSpecs[0].InstanceId != id {
+		return v, WrapErrorf(Error(GetNotFoundMessage("DdosbgpInstanceSpec", id)), NotFoundMsg, ProviderERROR)
+	}
+
+	v = resp.InstanceSpecs[0]
+	return v, WrapError(err)
+}
+
+func (s *DdosbgpService) UpdateDdosbgpInstanceName(instanceId string, name string, region string) error {
```

Comment:
There is no need a service method to modify instance name. Just move it to` resource update` is ok.